src/MongoDB/ReadPreference.c

```diff
@@ -80,28 +83,46 @@ PHP_METHOD(ReadPreference, __construct)
 			return;
 	}
 
-	switch(ZEND_NUM_ARGS()) {
-		case 2:
-			if (tagSets) {
-				bson_t *tags = bson_new();
+	if (tagSets) {
```

Comment:
Since I removed the `switch` statement and am processing `tagSets` first, I added a test that tag set exceptions are thrown before maxStalenessMS.